lib/src/main/scala/code/chymyst/jc/Molecules.scala

```diff
@@ -99,9 +99,9 @@ trait Molecule extends PersistentHashCode {
     *
     * @return {{{None}}} if the molecule emitter is not yet bound to any reaction site.
     */
-  private[jc] def consumingReactions: Option[Set[Reaction]] = reactionSiteOpt.map(_ => consumingReactionsSet)
+  private[jc] def consumingReactions: Option[List[Reaction]] = reactionSiteOpt.map(_ => consumingReactionsSet)
```

Comment:
I assume direct access to individual Reaction within consuming Reactions is unimportant and size is small (comment about choice of List). Since you started to introduce Array, I now see any instance of List in code with some suspicion.